CADETProcess/optimization/surrogate.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+
+from sklearn.gaussian_process import (
+    GaussianProcessRegressor, GaussianProcessClassifier)
+from sklearn.base import BaseEstimator
+
+from CADETProcess.optimization import (
+    Population, OptimizationProblem, OptimizationResults)
+
+class Surrogate:
+    def __init__(
+        self,
+        optimization_problem: OptimizationProblem,
+        population: Population
+    ):
+        self.optimization_problem = optimization_problem
+        self.surrogate_model_F: BaseEstimator = None
+        self.surrogate_model_G: BaseEstimator = None
+        self.surrogate_model_M: BaseEstimator = None
+        self.surrogate_model_CV: BaseEstimator = None
+        self.fit_gaussian_process(population)
+
+        # save a backup of bounds
+        self.lower_bounds_copy = optimization_problem.lower_bounds.copy()
+        self.upper_bounds_copy = optimization_problem.upper_bounds.copy()
+
+    def _reset_bounds_on_variables(self):
+        for var, lb, ub in zip(
+            self.optimization_problem.variables,
+            self.lower_bounds_copy,
+            self.upper_bounds_copy
+        ):
+            var.lb = lb
+            var.ub = ub
+
+    def fit_gaussian_process(self, population: Population):
+        X = population.x
+        F = population.f
+        G = population.g
+        M = population.m
+        CV = population.cv
+
+        gp_f = GaussianProcessRegressor()
+        gp_f.fit(X, F)
+        self.surrogate_model_F = gp_f
+
+        if G is not None:
+            gp_g = GaussianProcessRegressor()
+            gp_g.fit(X, G)
+            self.surrogate_model_G = gp_g
+
+        if M is not None:
+            gp_m = GaussianProcessRegressor()
+            gp_m.fit(X, M)
+            self.surrogate_model_M = gp_m
+
+        if CV is not None:
+            gp_cv = GaussianProcessClassifier()
+            gp_cv.fit(X, CV)
+            self.surrogate_model_CV = gp_cv
+
+
+
+    def estimate_objectives(self, X):
```

Comment:
I agree. But I wasn't sure about the design of this anyway. Maybe it would be better to design the functions/properties of Meta scores, constraint violations, ... analogous to what you did in the optimization problem / results? And only replace `evaluate` with `estimate`?

Reply:
Probably we should just mimic the `OptimizationProblem` interface for now. I.e. have individual functions for objectives, constraints, etc. This makes it easier to later fit it into existing architecture. 

Since at some point, I'd like to unify the `evaluate_` and evaluate_population` methods in the `OptimizationProblem` interface, I don't think we need to implement both here and already make the `evaluate` function work for both.